app/models/connected_paypal_account.rb

```diff
@@ -2,4 +2,50 @@
 
 class ConnectedPaypalAccount < ApplicationRecord
   has_one :competition_payment_integration, as: :connected_account
+
+  # TODO: Do we want to recycle Order items in PayPal the same way we recycle PaymentIntent items in Stripe?
+  def prepare_intent(registration, amount_iso, currency_iso, paying_user)
+    paypal_amount = PaypalRecord.amount_to_paypal(amount_iso, currency_iso)
+    params, paypal_order = PaypalInterface.create_order(self.paypal_merchant_id, paypal_amount, currency_iso)
+
+    paypal_record = PaypalRecord.create_from_api(paypal_order, 'paypal_order', params, self.paypal_merchant_id)
+
+    # memoize the payment intent in our DB because payments are handled asynchronously
+    # so we need to be able to retrieve this later at any time, even when our server crashes in the meantime…
+    PaymentIntent.create!(
+      holder: registration,
+      payment_record: paypal_record,
+      client_secret: paypal_record.paypal_id,
+      initiated_by: paying_user,
+      wca_status: paypal_record.determine_wca_status,
+    )
+  end
+
+  def capture_charges(payment_intent)
```

Comment:
The naming of this function is confusing because of it uses both "capture" and "charge". It being used across both payment providers is also confusing, because in Paypal we have to initiate the capture, and in Stripe we only have to record the charge - it implies an equivalence of functionality which I don't think exists